sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -663,3 +665,37 @@ def test_xfail_ignored_in_check_estimator():
     # Make sure checks marked as xfail are just ignored and not run by
     # check_estimator(), but still raise a warning.
     assert_warns(SkipTestWarning, check_estimator, NuSVC())
+
+
+def test_check_fit2d_1sample():
+
+    class MyEst(SVC):
+        # raises a bad error message when only 1 sample is passed
+        def fit(self, X, y):
+            if X.shape[0] == 1:
+                raise ValueError("non informative error message")
+
+    assert_raises_regex(
+        AssertionError,
+        "The error message should contain one of the following",
+        check_fit2d_1sample,
+        'estimator_name',
+        MyEst()
+    )
+
```

Comment:
Should we also be checking that with strict mode off check_fit2d_1sample passes without error?

Reply:
In the `test_strict_mode_...` tests in `test_common.py` we ensure that non-complient estimators still pass the suite when strict mode is off.

We don't test every single check though. I feel like it's not super necessary but no strong opinion.

Reply:
I would've thought it worthwhile to at least have one test of this error message checking functionality in relation to strict mode.

Reply:
We do have:

```py
    # value. This estimator would fail the check suite in strict mode,
    # specifically it would fail check_fit_non_negative
    def fit(self, X, y=None, **params):
        X = check_array(X, accept_sparse=('csr', 'csc'),
                        dtype=[np.float64, np.float32])
        try:
            check_non_negative(X, whom='')
        except ValueError:
            raise ValueError("Some non-informative error msg")

        return super().fit(X, y, **params)
```

and we make sure that it fails strictmode=True but passes strict_mode=False.

Would you rather have that for every check?